java/tests/src/test/java/org/whispersystems/libsignal/SessionCipherTest.java

```diff
@@ -95,7 +95,9 @@ public void testDecryptAfterReset() throws Exception {
     CiphertextMessage message2 = aliceCipher.encrypt(alicePlaintext);
 
     SessionRecord bobSession = bobStore.loadSession(aliceAddress);
+    bobSession.currentRatchetKeyMatches(Curve.generateKeyPair().getPublicKey()); // very unlikely but possible
```

Comment:
You appear to be missing an `assertFalse` here. I would then put the comment as the failure message:

```java
assertFalse("very unlikely but possible, just rerun", bobSession.currentRatchetKeyMatches(Curve.generateKeyPair().getPublicKey()));
```

If it ever did happen, we would see message and rerun tests.

Unless this is astronomically unlikely, in which case, forget the comment altogether.

Reply:
I had done this deliberately, but it really is astronomically unlikely. I'll put in the assert.